### PR TITLE
fix: Named CircuitBreakerApi should take ClassicActorSystemProvider

### PR DESCRIPTION
**Pull request by johanandren**
Binary breaking but source compatible change. ExtendedActorSystem was a really inconvenient parameter requiring an explicit cast in user code.

References https://github.com/akka/akka/issues/32307

### CODE REVIEW

**Review by octonato — APPROVED** [review]
LGTM, but waiting for a second opinion before merging because of compatibility issue.

File: akka-docs/src/main/paradox/project/migration-guide-2.8.x-2.9.x.md
Comment (patriknw) [inline akka-docs/src/main/paradox/project/migration-guide-2.8.x-2.9.x.md:66]: good to mention here

**Review by patriknw — APPROVED** [review]
LGTM
Really strange that it was ExtendedActorSystem

**Pull request merged**
